### ISSUE TITLE
App registration experience has changed at Microsoft.

### ISSUE THREAD

**Issue by lboening** [MarcelMeurer/PowerShellGallery-OneDrive#14]
I see the app registration experience is different at Microsoft. The readme.md would have to reflect that. Unfortunately, I am not familiar with that process yet and would be a poor choice for updating the documentation.

**Comment by tomko80** [comment]
Yes please somebody tell the steps needed to get this to work with having the personal  OneDrive and with the  new "experience" for setting up the auth.

**Comment by tomko80** [comment]
Here for anyone wondering over the same thing:

- Follow the steps outlined for the OneDrive for business and Azure
- When creating the registration select the last option: any org and personal
- Create "delegated" API permissions and grant Graph -> Files.ReadWrite.All
- Make sure you sign in with your personal account, not work or school
- Leave out any instance etc. attributes from the Powershell commands

Hope this helps.

**Comment by AgentMC** [comment]
Doesn't seem to work.  Trying to do it for personal 1drive.

If I call simply Get-ODAuthentication -ClientId 'client' -RedirectURI 'uri' -AppKey 'secret', it says "unauthorized_client: The client does not exist or is not enabled for consumers"

If I call it adding -ResourceId 'https://onedrive.live.com' it appeared not to be working, endlessly redirecting to login page.

Then I have added offline_access into Permissions list in Azure. And again called with -ResourceId 'https://onedrive.live.com'. This time it appeared to be working but no, after the login box disappeared, the console output is: ""error":"invalid_grant","error_description":"AADSTS65001: The user or administrator has not consented to use the application with ID 'clientid' named 'Access1Drive'. Send an interactive authorization request for this user and resource....." Meanwhile, Admin consent is granted, but nobody asked me for the user consent.

**Comment by AgentMC** [comment]
And one more scenario: if, when registering an app, I select "Applications from personal account" --> "Only associate with personal account" --> (o) Only Personal account

then the initial authentication process goes through, I can see a consent screen and even receive an email that "the new application is added to your account, it may read your data, blah", but then the get-ODAuthentication aniway fails with ""error":"invalid_request","error_description":"AADSTS9002331: Application
'clientid'(Test3) is configured for use by Microsoft Account users only. Please use the
/consumers endpoint to serve this request."

**Comment by lm-gh** [comment]
@AgentMC: I'm running into the exact same issue: "configured for use by Microsoft Account users only. Please use the /consumers endpoint to serve this request."

Did you already resolve it?

@tomko80: Releated to "delegated" API permissions and grant Graph -> Files.ReadWrite.All. I cannot add it since there is some administrator access required for this api.

**Comment by AgentMC** [comment]
@cbr-github kind of, the authentication sequence has to be done against https://login.microsoftonline.com/**consumers**/oauth2/v2.0/token not https://login.microsoftonline.com/**common**/oauth2/v2.0/token

But there is no way it seems I can pass the URL into the PS code, so I'm doing it through Postman. Surprisingly, after I have got through the whole OAUTH 2.0 sequence and got tokens, the https://api.onedrive.com/v1.0/ still does not accept it! The damn thing keeps saying 401 whatever I do! despite in the tab next to it I c an totally invoke refresh_token, so the data is all correct.


EDIT: OK, got it, despite MS docs literally say "for personal onedrive, use api.onedrive.com", this is wrong, you have to use https://graph.microsoft.com/v1.0/, e.g. https://graph.microsoft.com/v1.0/drives This hole thing is quite FUBAR TBH
